contrib/auto_annotations/build.gradle

```diff
@@ -9,7 +9,7 @@ description = 'OpenTelemetry Contrib Auto Annotations'
 ext.moduleName = "io.opentelemetry.contrib.auto.annotations"
 
 dependencies {
-    implementation project(':opentelemetry-api')
+    implementation project(':opentelemetry-api') exclude group: 'io.grpc', module: 'grpc-context'
```

Comment:
@bogdandrutu @carlosalberto any issues with this late-breaking change to minimize the dependencies of the annotations module?

Reply:
First request: Let's do it in a separate PR :)
Second request: I need to understand what you try to achieve :)

Reply:
1. sure thing; that's why I asked
2. @trask is concerned that people will need to pull in grpc-context just to use an annotation, and would prefer that not to be the case.

Reply:
I removed the change from this PR, FYI

Reply:
I removed the change from this PR, FYI